netsim/modules/bgp.py

```diff
@@ -604,6 +604,46 @@ def bgp_transform_community_list(node: Box, topology: Box) -> None:
   for s_type in list(clist.keys()):
     clist[s_type] = data.kw_list_transform(kw_xform,clist[s_type])
 
+"""
+link_ebgp_role_set -- set EBGP role for a regular link
+"""
+def ebgp_role_link(link: Box, topology: Box, EBGP_ROLE: str) -> None:
+  as_set = {}
+  for ifdata in link.get('interfaces',[]):                  # Collect BGP AS numbers from nodes
+    ndata = topology.nodes[ifdata.node]                     # ... connected to the link
+    node_as = ndata.get('bgp.as',None)
+    if node_as:
+      as_set[node_as] = True                                # ... and store them in a dictionary
+
+  if len(as_set) > 1 and not link.get("role"):              # If we have more than two AS numbers per link
```

Comment:
Should check this first, and not bother with collecting AS numbers if it's already set

Reply:
Done